transformer_engine/pytorch/float8_tensor.py

```diff
@@ -440,7 +440,9 @@ def transpose(
         dim0: int = 0,
         dim1: int = 1,
         *,
-        update_cache: str | bool = "reuse_only",
+        cache: bool = False,
+        update_cache: bool = True,
```

Comment:
I'd prefer lazy caching to be the default since `__torch_dispatch__` automatically resets the transpose when needed:
```suggestion
        update_cache: bool = False,
```